### PR TITLE
Save routes to manifest. Show routes in app details

### LINKED ISSUE #952: `epinio app manifest` should update routes
This is a fix for this comment: https://github.com/epinio/epinio/pull/942#issuecomment-958733119

Let's wait until #594 is done to avoid conflicts and fix it only once.

### PR DESCRIPTION
**Pull request by andreas-kupries**
Fix #952

### CODE REVIEW

**Comment by andreas-kupries** [comment]
check of latest CI issue in the api tests was all about BeforeEach and namespaces. ...
Rebased now, to the main having fixes in that area. Hopefully that fixes my problems here.

**Comment by andreas-kupries** [comment]
Given the CI :boom: for test case `list all apps across all namespaces` I have retested `-cli` locally.

|Variant|#Nodes|Result|
|:---|---:|:---|
|Just the failing test| 1 | :heavy_check_mark: |
|Entire `apps_test.go` suite| 1 | :heavy_check_mark: |
|Entire `apps_test.go` suite | 8 | :boom: |

Ok. Based on the above I conclude that we have cross-node test interference which breaks this particular test.
Or, IOW, when tests actually run concurrently some action of an unknown test X interferes with the failing test, breaking it.

:warning: __Note__ that this happens on a branch containing the recent fixes for targeting and listing namespaces.

Either this is something different, or the fix is not complete.

Regardless, the failing test is for `app list`, which is independent of the changes made by the PR (`app show`, `app manifest`).

As such I propose to disregard the failure and allow merging.

File: internal/cli/usercmd/app.go [hunk 1]
```diff
@@ -239,6 +239,14 @@ func (c *EpinioClient) AppShow(appName string) error {
...
 	}
 
+	msg = msg.WithTableRow("Routes", "")
+
+	if len(app.Configuration.Routes) > 0 {
+		for idx, route := range app.Configuration.Routes {
+			msg = msg.WithTableRow(fmt.Sprintf("  - %d", idx), route)
+		}
+	}
+
```
Comment (jimmykarily) [inline internal/cli/usercmd/app.go:249]: This information is already printed in lines 221-226:

```
✔️  Details:
|        KEY        |                 VALUE                 |
|-------------------|---------------------------------------|
| Status            | 1/1                                   |
| Username          | 81001e8dd6dac824                      |
| StageId           |                                       |
| Age               | 43s                                   |
| Restarts          |                                     0 |
| milliCPUs         |                                     0 |
| Memory            | 3.0 MiB                               |
| Routes            |                                       |
|                   | sample2.10.86.4.38.omg.howdoi.website |
| Desired Instances |                                     1 |
| Bound Services    |                                       |
| Environment       |                                       |
| Routes            |                                       |
|   - 0             | sample2.10.86.4.38.omg.howdoi.website |

```

**Review by jimmykarily — APPROVED** [review]

File: internal/cli/usercmd/app.go (hunk 1 above)
Comment (andreas-kupries) [reply internal/cli/usercmd/app.go:249]: @jimmykarily 

Not quite ... 221-226 shows the active routes, for when the app is running (pulled out of the Eorkload).
Only when the app is running. Nothing shown for app at rest.
My block shows the desired routes configured for the app (pulled out of the config). Even when the app is at rest.

Do wish to keep only one form ? If yes, which ?
If keeping both, we can simply fix the `KEY`s, i.e. `Active Routes`, vs `Desired Routes`.

I believe I would prefer that last.

Mixed mode: For running app show active, else show desired.

Comment (andreas-kupries) [reply internal/cli/usercmd/app.go:252]: And dd2e053605fb1f78f8d6c3491141cedb3086b223 implements mixed mode.

**Comment by andreas-kupries** [comment]
For the latest commit, first a created app, not deployed, second same app, deployed.

```
|        KEY        |                VALUE                |
|-------------------|-------------------------------------|
| Status            | not deployed                        |
| Desired Routes    |                                     |
|                   | fox.192.168.64.3.omg.howdoi.website |
| Desired Instances |                                   1 |
| Bound Services    |                                     |
| Environment       |                                     |
```

```
|        KEY        |                VALUE                |
|-------------------|-------------------------------------|
| Status            | 1/1                                 |
| Username          | bce349538822b1c8                    |
| StageId           |                                     |
| Age               | 5m17s                               |
| Restarts          |                                   0 |
| milliCPUs         |                                   0 |
| Memory            | 19.3 MiB                            |
| Active Routes     |                                     |
|                   | fox.192.168.64.3.omg.howdoi.website |
| Desired Instances |                                   1 |
| Bound Services    |                                     |
| Environment       |                                     |
```

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**